x-pack/plugins/security_solution/public/timelines/components/timeline/tabs/shared/use_timeline_control_columns.tsx

```diff
@@ -84,49 +138,7 @@ export const useTimelineControlColumn = ({
             />
           );
         },
-        rowCellRender: (
-          props: EuiDataGridCellValueElementProps & UnifiedTimelineDataGridCellContext
-        ) => {
-          /*
-           * In some cases, when number of events is updated
-           * but new table is not yet rendered it can result
-           * in the mismatch between the number of events v/s
-           * the number of rows in the table currently rendered.
-           *
-           * */
-          if ('rowIndex' in props && props.rowIndex >= events.length) return <></>;
-          props.setCellProps({
-            className:
-              props.expandedEventId === events[props.rowIndex]?._id
-                ? 'unifiedDataTable__cell--expanded'
-                : '',
-          });
-
-          return (
-            <TimelineControlColumnCellRender
-              rowIndex={props.rowIndex}
-              columnId={props.columnId}
-              timelineId={timelineId}
-              ariaRowindex={props.rowIndex}
-              checked={false}
-              columnValues=""
-              data={events[props.rowIndex].data}
-              ecsData={events[props.rowIndex].ecs}
-              loadingEventIds={EMPTY_STRING_ARRAY}
-              eventId={events[props.rowIndex]?._id}
-              index={props.rowIndex}
-              onEventDetailsPanelOpened={noOp}
-              onRowSelected={noOp}
-              refetch={refetch}
-              showCheckboxes={false}
-              setEventsLoading={noOp}
-              setEventsDeleted={noOp}
-              pinnedEventIds={pinnedEventIds}
-              eventIdToNoteIds={eventIdToNoteIds}
-              toggleShowNotes={onToggleShowNotes}
-            />
-          );
-        },
+        rowCellRender: JEST_ENVIRONMENT ? RowCellRender : React.memo(RowCellRender),
```

Comment:
I'm not a huge fan of checking this here. Could you elaborate in a comment here on why it's necessary to not memoize in test mode?

Reply:
if we do not do it, we get below warning in JEST tests. EUI is not happy when a memoized component is passed.

![grafik](https://github.com/user-attachments/assets/eae7062e-b4d7-4c0f-bc15-ad3f66a47a4b)

Reply:
if you could say that in a comment right there, that would be perfect 👍 
Plus, we should probably open up an issue with EUI about this. Sounds like one of their prop types is wrong.